BUILDING-cmake.md

```diff
@@ -6,22 +6,37 @@ This file contains in-depth information for building with the CMake build system
 
 Want to build it fast?
 
-Required tools: `cmake`. 
+Required tools and dependencies:
+
+- CMake 3.20 or higher.
+- A working toolchain, e.g. Visual Studio on Windows or the `build-essentials` package on Ubuntu Linux.
+- Main OpenGL libraries and development files.
+- The `GLEW` Library on Windows.
+
+To use the library in other projects, it is required to install it. Use `CMAKE_INSTALL_PREFIX` to specify the
+installation directory.
+
+From the project root, execute:
 
 ```shell
 mkdir build
 cd build
-cmake ..
-make -j8
+cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/path/to/install-dir
+cmake --build . --target install --config Release
 ```
 
-Output binaries will be in `build/src/...`.
+If the build succeeded, you should now have the projectM libraries and include files in the specified install dir.
```

Comment:
Maybe it'd be simpler to not tell people to set the install prefix and use the system install prefix? Wouldn't that simplify this quite a bit?

Reply:
Depends on the use case, actually. If you're using libprojectM for development, and probably want to have different variants (static/shared) etc., installing somewhere into the user directory is to be preferred. Besides that, installing into the system's default prefix requires admin/root privileges. Plus, the install mechanism doesn't care about existing files, so users may unexpectedly end up overwriting files of their system, which - on Linux - can seriously mess up the package manager's file database.

So in general, setting the prefix to something less destructible is preferable.

Even in package building and on meta distros like Gentoo, files are installed into a staging dir before being copied into their final locations.

Additionally, when using CPack, `CMAKE_INSTALL_PREFIX` is overridden to CPack's temporary install dir anyway.